libr/core/cmd_write.c

```diff
@@ -2017,8 +2017,29 @@ static int cmd_write(void *data, const char *input) {
 		wc_handler_old (core, input + 1);
 		break;
 	case ' ': // "w"
-		w_handler_old (core, input + 1);
+	{
+		size_t len = core->blocksize;
+		const char *current_charset = r_config_get (core->config, "cfg.charset");
+		if (R_STR_ISEMPTY (current_charset)) {
+			w_handler_old (core, input + 1);
```

Comment:
imho the logic of text encoding should be done in a diferent place. see the w_handle_old code and make the encoding there, not in every caller

Reply:
@trufae I did same for the command ps. I might change ps too.

Reply:
yes